torch_xla/csrc/runtime/pjrt_compile_only.h

```diff
@@ -0,0 +1,48 @@
+#ifndef XLA_CLIENT_PJRT_COMPILE_ONLY_H_
+#define XLA_CLIENT_PJRT_COMPILE_ONLY_H_
+
+#include "xla/pjrt/pjrt_client.h"
+#include "xla/pjrt/pjrt_future.h"
+
+namespace xla {
+
+class CompileOnlyPjRtBuffer final : public PjRtBuffer {
```

Comment:
Thanks for reducing the dependency on PJRT. Why do you still need an implementation of PjRtBuffer? Could `PjRtCompilationClient::PjRtData` just not hold a buffer?

Reply:
I have tried to use PjRtData without a buffer before. The issue is that Pytorch will produce the correct IR but the HLO is incorrect from lowering_ctx.BuildXla(). This is due to all tensors being mapped to the same null buffer and the HLO will get rid of the redundant tensors with the same null buffer. The solution is to create our own `PjRtCompileOnlyBuffer` that doesn’t interact with the real buffer. As long as our buffer pointers are not null and pointing to different addresses, the HLO will be produced correctly.

The details are mentioned in the `AOT compilation` design doc. However, I think I can further simplify the buffer class in this case and move them inside the pjrt_compilation_client.h.

Reply:
I just simplified the PjRtBuffer into :
```
struct Buffer {
  xla::Shape shape;
  Buffer(xla::Shape shape): shape(shape) {}
};
```
This way we can get rid off the openxla buffer decenency while still maintaining the PjRtData buffer attribute.